### PR TITLE
adding coveralls

### PR DESCRIPTION
**Pull request by fooelisa**
Added coveralls integration to see if this is something we would like to have.

### CODE REVIEW

**Comment by mirceaulinic** [comment]
https://github.com/napalm-automation/napalm-base/pull/91 will probably increase with 1-2% the coverage :P

**Comment by dbarrosop** [comment]
I really want this, howerver, bear in mind that for us, functional/acceptance tests are more important than unit tests and coverage.

**Comment by fooelisa** [comment]
Can't seem to be able to move this to pytest instead of nose, it doesn't pick up the tests at all so it seems. Could you take a look @dbarrosop, what's missing here?

**Comment by ogenstad** [comment]
Try: `py.test --cov-report term-missing test/` in .travis.yml

**Comment by fooelisa** [comment]
No `missing-term` will change the report as to displaying missing lines or not, however pytest is not actually finding/starting the tests at all here (which it does in eos). I think they have to be called "test_*" to be picked up.

**Comment by ogenstad** [comment]
Ah ok. Yes pytest needs the tests to begin with test_

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,9 @@ python:
 
 install:
   - pip install .
+  - pip install -r requirements-dev.txt
   - pip install -r test/unit/requirements.txt
+  - pip install coveralls
```
Comment (dbarrosop) [inline .travis.yml]: Why don't you just add it to `requirements-dev.txt`?

File: .travis.yml [hunk 2]
```diff
@@ -17,6 +19,7 @@ deploy:
     branch: master
 
 script:
-  - nosetests ./test/unit/TestGetNetworkDriver.py
-  - nosetests ./test/unit/TestHelpers.py
-  - nosetests ./test/unit/TestNapalmTestFramework.py
+  - py.test --cov-report= test/
+  # - pylama .
```
Comment (dbarrosop) [inline .travis.yml]: Why are commenting this one?

File: .travis.yml (hunk 1 above)
Comment (fooelisa) [inline .travis.yml]: No reason, would you prefer it there?

File: .travis.yml (hunk 2 above)
Comment (fooelisa) [inline .travis.yml]: leftover from my testing, pylama was producing a lot of output for base

**Comment by fooelisa** [comment]
oh now I remember, pylama was erroring, did you see this in other modules? -eos looks set up the same to me 

`napalm_base/__init__.py WARNING: __all__ is defined as a list, this means pydocstyle cannot reliably detect contents of the __all__ variable, because it can be mutated. Change __all__ to be an (immutable) tuple, to remove this warning. Note, pydocstyle uses __all__ to detect which definitions are public, to warn if public definitions are missing docstrings. If __all__ is a (mutable) list, pydocstyle cannot reliably assume its contents. pydocstyle will proceed assuming __all__ is not mutated.`

File: .travis.yml (hunk 1 above)
Comment (dbarrosop) [inline .travis.yml]: Yes, please

**Comment by dbarrosop** [comment]
This has been broken for a while, closing it. Please, reopen and fix when you have the time.

**Pull request closed without merging**